### ISSUE TITLE
Question about memory / garbage collection (linux) slows down after 24 hrs

### ISSUE THREAD

**Issue by DrStein99** [fireice-uk/xmr-stak#1193]
I am comparing my stats between all my machines and can not dismiss how convenient it is to notice how they are so very productive right after a fresh reboot - and when I check after 24 hours, the machines slow down.  I do not have precise numbers yet, but I do have SOMETHING.

I noticed when I started ./xmr-stak at command line instead of run as service, and check my log.  Initially after I execute just after a fresh new reboot - the whole INIT process flashes at fastest possible speed across my console.  After 24 hours (does not need 24, probably just a few hours) if I exit XMR-STAK, at the console.  Wait 30 seconds, and then execute ./xmr-stak - the INIT process comes across the console in slow motion, like as if the machine is under some type of load.  I checked in TOP, I could not find anything obvious (with my novice level TOP reading skills).

So what I have done, was add some lines of code to time the boot up process.  (Please forgive my crude cut-and-paste style documentation, I have not yet learned how to do the fancy git-hub tricks to compare files and show the highlights)

minethd.cpp : 
```
if( !jconf::inst()->parse_config() )
	{
		win_exit();
	}
	// 3/24/18 - DrStein99: This function records the time from this point to log how fast the machine is ready
	size_t iBootStartTime = get_timestamp_ms();

	//Launch the requested number of single and double threads, to distribute
	//load evenly we need to alternate single and double threads
	size_t iTMP, itmpThreadCount = jconf::inst()->GetThreadCount();
	pvThreads.reserve( itmpThreadCount );
```

(further down the code)

```
	// 3/24/2018 - DrStein99 
	size_t iBootTimeMS = get_timestamp_ms() - iBootStartTime;
	printer::inst()->print_msg(L1, "BOOT TIME           | %4zums",
		iBootTimeMS);
	
	return pvThreads;

```
This statistic will report number of milliseconds for the machine to perform that procedure.

So what I come up with on my #1 Windows (2) E5-2670 machine, it reports "62ms" to do the "BOOT TIME" to allocate thread memory, after I have had this machine running 24/7 with many other things on and off.

On the Linux machine with (4) E7-4850 processors after being run for 10 straight hours, I exit xmr-stak just to re-compile and start the process.  BOOT TIME returns "2381ms" on the 1st machine, and "1124ms" on the 2nd machine with nearly identical hardware besides 32gb additional system ram.  

After rebooting E7 #1, BOOT TIME is "2307ms".  After rebooting E7 #2, BOOT TIME is "1211ms". 

On Linux machine #3 with (2) E5-2450L processors & 16gb ram, BOOT TIME is "371ms" after running for freshly rebooted.  I run for 10 entire minutes, shut down.  Then start xmr-stak again, the BOOT TIME is "533ms".

---
So in conclusion, I have to say - that of course there is SOMETHING going on SOMEWHERE to slow the machine down, and I am uncertain what it is or where to look for next.  I was hoping anyone else could just read my notes and speculate on any theories, or where I could go to nail down any parasitic cause that could interfere with the whole process of this application.

**Comment by jacobgmartin** [comment]
Same here!

**Comment by fireice-uk** [comment]
There is no gc in c++. If a program leaks memory you will see it taking up gigabytes after a couple of days,

**Comment by DrStein99** [comment]
I just wanted to add a note; the BOOT TIME increases each time I restart xmr-stak, by 100-200ms.  It continues to do this until I hot-reboot the machine.  At this point, I feel a regular reboot schedule should probably be implemented, with cron-jobs to restart xmr-stak probably 4-8 hours.  It is the only thing I know that works when I analyze my hash-rate from the pool and see the graph slide down over these periods of time.
